components/sup/src/manager/service/hook_runner.rs

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) 2019 Chef Software Inc. and/or applicable contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Runs a service lifecycle hook on a separate thread, and wraps the
+//! whole execution in a future.
+//!
+//! Ideally, we'd want to use something like
+//! [tokio_process](https://github.com/alexcrichton/tokio-process),
+//! but we're not able to use that based on how our Windows hooks get
+//! executed. If that were to be re-cast in terms of Rust's
+//! `std::process::Command`, we could consider it. In the meantime,
+//! this seems to do the trick.
+
+use super::{hook_timer, spawned_future::SpawnedFuture, Pkg};
+use crate::error::SupError;
+use futures::{sync::oneshot, IntoFuture};
+use habitat_common::templating::hooks::Hook;
+use habitat_core::service::ServiceGroup;
+use std::{
+    sync::{Arc, Mutex},
+    thread,
+};
+
+pub struct HookRunner<H: Hook> {
+    hook: Arc<Mutex<H>>,
+    sg: ServiceGroup,
```

Comment:
```suggestion
    service_group: ServiceGroup,
```
This brevity seems excessive; similarly elsewhere

Reply:
Probably also worthwhile to use `service_group` if we're moving to implicit field struct creation pattern (e.g. `HookRunner{hook, service_group, ... }`)